lua/nvim-tree/renderer/builder.lua

```diff
@@ -364,13 +406,44 @@ end
 
 function Builder:build(tree, unloaded_bufnr)
   local num_children = self:_get_nodes_number(tree.nodes)
+  local diagnostic_severities_by_path = diagnostics.get_diagnostics()
```

Comment:
This doesn't fit in here; it should go back in diagnostics.lua.

I like `_get_diagnostic_icon` - nice and consistent.

Proposal:

- Move this back into `diagnostics.update`
- `diagnostics.update` records the diagnostic state of the node. Something like `git_status`
- call draw following `diagnostics.update`
- `_get_diagnostic_icon` calls something like `local diagnostic_icon = diagnostics.get_icon(node)`